src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs

```diff
@@ -464,21 +464,35 @@ private static IEnumerable<ITestInfo> GetTestMethodInfosForMethod(IMethodSymbol
         List<AttributeData> filterAttributes = new();
         foreach (var attr in method.GetAttributesOnSelfAndContainingSymbols())
         {
+            var hasSkip = attr.NamedArguments.Any(x => x.Key == "Skip");
+
             switch (attr.AttributeClass?.ToDisplayString())
             {
                 case "Xunit.ConditionalFactAttribute":
-                    filterAttributes.Add(attr);
-                    factAttribute = true;
+                    if (!hasSkip)
```

Comment:
This seems like a prime example for the newer C# switch syntax

```
case "Xunit.ConditionalFactAttribute" when !hasSkip:
```
though perhaps the right solution is to put all of these into `filterAttributes` so that filtering can then drop them like other filters do (and skip reporting, etc., would have a single place to be implemented).